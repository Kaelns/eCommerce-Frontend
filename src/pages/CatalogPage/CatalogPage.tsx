import type { SxStylesMap } from '@/shared/model/types';

import { Box, Tooltip } from '@mui/material';
import ClearIcon from '@mui/icons-material/Clear';
import { Stack, useTheme, useMediaQuery } from '@mui/system';

import { CatalogProducts } from '@/pages/CatalogPage/ui/layout/CatalogProducts';
import { CatalogSideDrawer } from '@/pages/CatalogPage/ui/layout/CatalogSideDrawer';
import { setIsOpenFilterDrawerAction } from '@/pages/CatalogPage/model/catalogPage.slice';
import { OpenCatalogSideDrawerBtn } from '@/pages/CatalogPage/ui/components/OpenCatalogSideDrawerBtn';

import { CatalogSearch, CatalogFilterForm, CategoriesBreadcrumb } from '@/features/catalog-filters';

import { ContainedIconBtn } from '@/shared/ui/elements';
import { useAppDispatch } from '@/shared/lib/redux';

const sxStyles: SxStylesMap = {
  container: {
    mt: -2
  },
  filterForm: {
    maxWidth: 400
  },
  drawerContainer: {
    overflowY: 'scroll',
    p: 3,
    pb: 0
  },
  drawerCloseBtn: {
    position: 'fixed',
    top: '1rem',
    right: '2rem',
    zIndex: 1000
  }
};

export function CatalogPage() {
  const theme = useTheme();
  const dispatch = useAppDispatch();

  const isMatchesLaptopBig = useMediaQuery(theme.breakpoints.up('laptopBig'));

  const handleCloseSideDrawer = () => {
    dispatch(setIsOpenFilterDrawerAction(false));
  };

  return (
    <Box sx={sxStyles.container}>
      <CategoriesBreadcrumb />
      <Stack component="section" gap={2} mt={1}>
        <Stack direction="row" gap={1}>
          <CatalogSearch />
          <OpenCatalogSideDrawerBtn />
        </Stack>

        <Stack direction="row" gap={2}>
          {isMatchesLaptopBig && <CatalogFilterForm component="aside" flex={2} sx={sxStyles.filterForm} />}
          <CatalogProducts flex={8} />
        </Stack>

        <CatalogSideDrawer>
          <Tooltip title="Close side menu">
            <ContainedIconBtn onClick={handleCloseSideDrawer} sx={sxStyles.drawerCloseBtn}>
              <ClearIcon />
            </ContainedIconBtn>
          </Tooltip>

          <Stack gap={2} sx={sxStyles.drawerContainer}>
            <CatalogFilterForm component="aside" sx={sxStyles.filterForm} />
          </Stack>
        </CatalogSideDrawer>
      </Stack>
    </Box>
  );
}
