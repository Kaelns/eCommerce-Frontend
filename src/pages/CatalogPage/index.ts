export { CatalogPage } from '@/pages/CatalogPage/CatalogPage';
export { prefetchCatalogPageLoader } from '@/pages/CatalogPage/model/prefetchCatalogPage.loader';
