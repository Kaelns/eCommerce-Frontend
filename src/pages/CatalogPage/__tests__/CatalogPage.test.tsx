import { CatalogPage } from '@/pages/CatalogPage/CatalogPage';
import { render } from '@testing-library/react';

describe('Given CatalogPage component: ', () => {
  it('when rendered, should match snapshot', () => {
    const { asFragment } = render(<CatalogPage />);
    expect(asFragment()).toMatchSnapshot();
  });
});
