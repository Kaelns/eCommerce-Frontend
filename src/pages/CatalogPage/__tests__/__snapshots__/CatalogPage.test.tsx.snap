// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Given CatalogPage component:  > when rendered, should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-1ytub8b"
  >
    <nav
      class="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root css-8yvas4-MuiTypography-root-MuiBreadcrumbs-root"
    >
      <ol
        class="MuiBreadcrumbs-ol css-4pdmu4-MuiBreadcrumbs-ol"
      >
        <li
          class="MuiBreadcrumbs-li"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary Mui-disabled MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-13tp3gi-MuiButtonBase-root-MuiButton-root"
            disabled=""
            tabindex="-1"
            type="button"
          >
            No category
          </button>
        </li>
      </ol>
    </nav>
    <div
      class="MuiStack-root css-d25iif-MuiStack-root"
    >
      <div
        class="MuiStack-root css-1oxswqj-MuiStack-root"
      >
        <div
          class="MuiBox-root css-1new6zd"
        >
          <div
            class="MuiStack-root css-8xhq8x-MuiStack-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-18w7uxr-MuiSvgIcon-root"
              data-testid="SearchIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14"
              />
            </svg>
          </div>
          <div
            class="MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-adornedEnd css-18j0p5e-MuiInputBase-root"
          >
            <input
              class="MuiInputBase-input MuiInputBase-inputAdornedEnd css-1jrstcx-MuiInputBase-input"
              placeholder="Searchâ€¦"
              type="text"
              value=""
            />
            <button
              class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-sizeMedium MuiInputAdornment-root MuiInputAdornment-positionEnd css-1b5u5pl-MuiButtonBase-root-MuiIconButton-root-MuiInputAdornment-root"
              disabled=""
              tabindex="-1"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-18w7uxr-MuiSvgIcon-root"
                data-testid="CloseIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                />
              </svg>
            </button>
          </div>
        </div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-1x76y4a-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          <div
            class="MuiStack-root css-1uhex6k-MuiStack-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-18w7uxr-MuiSvgIcon-root"
              data-testid="FilterListIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10 18h4v-2h-4zM3 6v2h18V6zm3 7h12v-2H6z"
              />
            </svg>
            <p
              class="MuiTypography-root MuiTypography-body1 css-b3z1-MuiTypography-root"
            >
              Filters
            </p>
          </div>
        </button>
      </div>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-row MuiGrid-spacing-xs-2 css-1u5xnsm-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-row MuiGrid-grid-tablet-grow css-1d90wfh-MuiGrid-root"
        >
          <div
            class="MuiBox-root css-1kxonj9"
          >
            <div
              class="MuiStack-root css-r78jfo-MuiStack-root"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave css-1lbmg9-MuiSkeleton-root"
              />
              <span
                class="MuiSkeleton-root MuiSkeleton-circular MuiSkeleton-wave css-7bduwa-MuiSkeleton-root"
              />
              <span
                class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave css-1lbmg9-MuiSkeleton-root"
              />
              <span
                class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave css-1lbmg9-MuiSkeleton-root"
              />
              <span
                class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave css-13v5yvq-MuiSkeleton-root"
              />
              <span
                class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave css-13v5yvq-MuiSkeleton-root"
              />
              <span
                class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave css-13v5yvq-MuiSkeleton-root"
              />
              <span
                class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave css-13v5yvq-MuiSkeleton-root"
              />
              <span
                class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave css-13v5yvq-MuiSkeleton-root"
              />
            </div>
            <div
              class="MuiStack-root css-x6bnna-MuiStack-root"
            >
              <h5
                class="MuiTypography-root MuiTypography-h5 css-1yej70y-MuiTypography-root"
              >
                There is no products
              </h5>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
